router/router.go

```diff
@@ -700,6 +741,56 @@ func init() {
 	loadConfig()
 }
 
+func (rt *HandleT) backendConfigSubscriber() {
+	ch := make(chan utils.DataEvent)
+	backendconfig.Subscribe(ch, "backendConfig")
+	for {
+		config := <-ch
+		configSubscriberLock.Lock()
+		allSources := config.Data.(backendconfig.SourcesT)
+
+		// Producers of stream destinations are created or closed while server starts up or workspace config gets changed
+		for _, source := range allSources.Sources {
+			if len(source.Destinations) > 0 {
+				for _, destination := range source.Destinations {
+					if destination.DestinationDefinition.Name == rt.destID && misc.ContainsString(customDestinations, rt.destID) {
+						key := source.ID + "-" + destination.ID
+						destConfig := destination.Config
+						if destinationConfigProducerMap[key] != (ProducerConfig{}) {
+							producer := destinationConfigProducerMap[key].Producer
+							config := destinationConfigProducerMap[key].Config
+							if reflect.DeepEqual(config, destConfig) {
+								if !destination.Enabled {
+									logger.Infof("======== closing existing producer as destination disabled for destination: %v and source: %v", destination.Name, source.Name)
+									customdestinationmanager.CloseProducer(producer, rt.destID)
+									delete(destinationConfigProducerMap, key)
+								}
+								continue
+							}
+							logger.Infof("=========config changed ======== closing existing producer ======= for destination: %v and source: %v", destination.Name, source.Name)
+							customdestinationmanager.CloseProducer(producer, rt.destID)
+							delete(destinationConfigProducerMap, key)
+						}
+						if destination.Enabled {
+							producer, err := customdestinationmanager.GetProducer(destConfig, rt.destID)
+							producerConfig := ProducerConfig{Config: destConfig, Producer: producer}
+							destinationConfigProducerMap[key] = producerConfig
+							if err == nil {
+								logger.Infof("========== created new producer: %v for destination: %v and source %v", producer, destination.Name, source.Name)
+							} else {
+								producerConfig := ProducerConfig{Config: destConfig, Producer: producer}
+								destinationConfigProducerMap[key] = producerConfig
```

Comment:
We need to synchronize the access to destinationConfigProducerMap. Here we are writing and reading inside workerProcess.

Reply:
Used the same mutex.